File: Quiz.java
import java.util.Scanner;

public class Quiz {
    // VARIABELEN (ALTIJD BENADERBAAR)
    // Woordenlijst (vul aan tot 100 voor meer variatie)
    private static final String[][] woordenlijst = {
        {"kat", "gato"},
        {"auto", "coche"},
        {"hond", "perro"},
        {"huis", "casa"},
        {"boek", "libro"},
        // ... meer woorden
    };
    
    private static int aantalVragen = 5; // default
    private static int score = 0;
    private static Scanner scanner = new Scanner(System.in);
    
    // FUNCTIES
    public static void startQuiz() {
        System.out.println("🎉 Welkom bij de Nederlandse-Spaanse Woordquiz! ");
        System.out.println("Je krijgt " + aantalVragen + " Nederlandse woorden.");
        System.out.println();
        stelVraag(1);
    }
    
    public static void stelVraag(int vraagNummer) {
        if (vraagNummer > aantalVragen) {
            System.out.println("Quiz klaar! Je hebt " + score + " van de " + aantalVragen + " goed.");
            scanner.close();
            return;
        }
        
        String nederlandsWoord = woordenlijst[vraagNummer - 1][0];
        String spaansWoord = woordenlijst[vraagNummer - 1][1];
        
        System.out.print("Vraag " + vraagNummer + ": Wat is het Spaanse woord voor '" + nederlandsWoord + "'? ");
        String antwoord = scanner.nextLine().trim();
        
        if (antwoord.equals(spaansWoord)) {
            System.out.println("✅ Goed gedaan!");
            score = score + 1;
        } else {
            System.out.println("❌ Helaas, het juiste antwoord was '" + spaansWoord + "'.");
        }
        
        System.out.println("Score: " + score);
        System.out.println();
        
        stelVraag(vraagNummer + 1);
    }
    
    public static void main(String[] args) {
        // UITVOERENDE CODE
        // Vraag eerst het aantal vragen
        System.out.print("Hoeveel vragen wil je? ");
        String input = scanner.nextLine();
        
        try {
            int aantal = Integer.parseInt(input);
            if (aantal > 0 && aantal <= woordenlijst.length) {
                aantalVragen = aantal;
            } else {
                System.out.println("Ongeldig aantal, standaardwaarde (" + aantalVragen + ") wordt gebruikt.");
            }
        } catch (NumberFormatException e) {
            System.out.println("Ongeldig aantal, standaardwaarde (" + aantalVragen + ") wordt gebruikt.");
        }
        
        startQuiz();
    }
} 

File: quiz.js
const readline = require('readline');

// VARIABELEN (ALTIJD BENADERBAAR)
// Woordenlijst (vul aan tot 100 voor meer variatie)
const woordenlijst = [
    ["kat", "gato"],
    ["auto", "coche"],
    ["hond", "perro"],
    ["huis", "casa"],
    ["boek", "libro"],
    // ... meer woorden
];

let aantalVragen = 5; // default
let score = 0;

// FUNCTIES
function startQuiz() {
    console.log("🎉 Welkom bij de Nederlandse-Spaanse Woordquiz! ");
    console.log(`Je krijgt ${aantalVragen} Nederlandse woorden.`);
    console.log("");
    stelVraag(1);
}

function stelVraag(vraagNummer) {
    if (vraagNummer > aantalVragen) {
        console.log(`Quiz klaar! Je hebt ${score} van de ${aantalVragen} goed.`);
        return;
    }
    const nederlandsWoord = woordenlijst[vraagNummer - 1][0];
    const spaansWoord = woordenlijst[vraagNummer - 1][1];

    const rlStelVraag = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });

    const vraag = `Vraag ${vraagNummer}: Wat is het Spaanse woord voor '${nederlandsWoord}'? `;
    rlStelVraag.question(vraag, (antwoord) => {
        if (antwoord && antwoord.trim() === spaansWoord) {
            console.log("✅ Goed gedaan!");
            score = score + 1;
        } else {
            console.log(`❌ Helaas, het juiste antwoord was '${spaansWoord}'.`);
        }
        console.log(`Score: ${score}`);
        console.log("");
        rlStelVraag.close();
        stelVraag(vraagNummer + 1);
    });
}

// UITVOERENDE CODE
// Vraag eerst het aantal vragen
const rlAantal = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rlAantal.question("Hoeveel vragen wil je? ", (input) => {
    const aantal = parseInt(input, 10);
    if (!isNaN(aantal) && aantal > 0 && aantal <= woordenlijst.length) {
        aantalVragen = aantal;
    } else {
        console.log(`Ongeldig aantal, standaardwaarde (${aantalVragen}) wordt gebruikt.`);
    }
    rlAantal.close();
    startQuiz();
});

